doc/macsec-gearbox/PAI-Proposal-Failover.md

```diff
@@ -0,0 +1,381 @@
+![](RackMultipart20200925-4-s340pz_html_6e4eda26a3e5fd14.png)
+
+# External PHY Failover
+
+# Proposal
+
+| **Title** | **External PHY Abstraction Interface** |
+| --- | --- |
+| **Authors** | **Broadcom** |
+| **Status** | **In review** |
+| **Type** | **Standards track** |
+| **Created** | **8/31/2020** |
+| **SAI-Version** | **0.1** |
+
+
+# Overview
+
+The purpose of this document is to describe PHY failover (FO) functionality and interface to manage failover functionality of the PHY. PHY is the connector between MAC (SerDes) and physical medium such as optical fiber or copper transceivers. PHY failover feature uses two physical port configurations: primary and secondary. This can be enabled on system/host or line/client side of a port. It provides the ability to switch from primary port to secondary port or vice versa without losing data/link.
+
+![](RackMultipart20200925-4-s340pz_html_6fd49d19699540a6.png)
+
+In the above multi-port configuration, failover mode is configured on the system side of external PHY. Primary and secondary ports are physical connections between MAC and PHY. The FO configuration uses primary and secondary port connections on selected side (system side or line side).
+
+The PHY failover mode enables a port to broadcast data on both primary and secondary ports. The MAC would be configured to receive the data on either primary or secondary ports. PHY failover active ingress port can be configured as either primary or secondary port. When the link between MAC and PHY goes down on primary port, user can switch to secondary port.
+
+## Failover Configuration on PHY
+
+Failover feature can be configured on all ports of the PHY. Failover ports can operate in either hitless (the ability to switch over port without losing link) or non-hitless mode.
+
+Additions to saiport.h
+
+/\*\*
```

Comment:
can you use code block to make it clear.

Reply:
Added block